root/context.js

```diff
@@ -0,0 +1,26 @@
+/*
+ * @flow
+ * Copyright (C) 2018 MetaBrainz Foundation
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+import React from 'react';
+import type {ComponentType} from 'react';
+
+// $FlowFixMe
+export const CatalystContext = React.createContext(null);
+
+type ContextPropT = {$c: CatalystContextT};
+
+export function withCatalystContext<P: ContextPropT, T: $Diff<P, ContextPropT>>(
```

Comment:
See https://reactjs.org/docs/context.html#consuming-context-with-a-hoc which follows the same pattern.

It's an HOC (higher-order component) that takes a component which needs a `$c`, wraps it in `<CatalystContext.Consumer />`, and passes the `$c` to it.